clang/include/clang/Serialization/ASTWriter.h

```diff
@@ -661,8 +661,8 @@ class ASTWriter : public ASTDeserializationListener,
 
   /// Write a precompiled header for the given semantic analysis.
   ///
-  /// \param SemaRef a reference to the semantic analysis object that processed
-  /// the AST to be written into the precompiled header.
+  /// \param Subject The object that processed the input to be written into the
+  /// AST file.
```

Comment:
The first line of the doc comment above also mentions semantic analysis.  I wonder if it would help to spell out the use case more explicitly and say something along these lines:

Write a precompiled header or module for an AST with the given semantic analysis, or a dependency scanner module for an input using the given preprocessor.

\param Subject The semantic analysis object that processed the AST to be written, or in the case of a dependency scanner module the preprocessor that processed the input.

Reply:
Sounds good, I clarified the documentation a bit, WDYT now?